extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -149,6 +184,12 @@ public AgroalDataSource doCreateDataSource(String dataSourceName) {
                     "Unable to load the datasource driver " + resolvedDriverClass + " for datasource " + dataSourceName, e);
         }
 
+        if (dataSourceJdbcBuildTimeConfig.tracing && dataSourceJdbcRuntimeConfig.tracing.enabled.isPresent() &&
```

Comment:
You could probably reuse the local variable set previously if you made it visible?